lib/cylc/task_pool.py

```diff
@@ -1282,18 +1290,18 @@ def process_queued_db_ops(self):
 
     def force_spawn(self, itask):
         """Spawn successor of itask."""
-        if itask.state.has_spawned():
+        if itask.has_spawned:
             return None
-        itask.state.set_spawned()
+        itask.has_spawned = True
         itask.log(DEBUG, 'forced spawning')
-        new_task = itask.spawn('waiting')
+        new_task = itask.spawn(TASK_STATUS_WAITING)
         if new_task and self.add_to_runahead_pool(new_task):
             return new_task
         else:
             return None
 
-    def spawn_tasks(self):
-        """Spawn successors of tasks in pool.
+    def spawn_all_tasks(self):
+        """Spawn successors of tasks in pool, if they're ready.
```

Comment:
(Sorry, it appears I forgot to address this comment). No, it doesn't change spawning behaviour at all.  The specific change here merely makes the method name and comment more descriptive of what's actually happening (we iterate through *all* task proxies and spawn only those that are *ready* to spawn).  More generally, as commented above, the 'has_spawned' boolean is now a simple task proxy attribute, instead of being held inside the class that handles status strings like 'waiting' etc.  No change in behaviour though.